src/server/cluster/cluster_family.cc

```diff
@@ -51,6 +52,49 @@ thread_local shared_ptr<ClusterConfig> tl_cluster_config;
 
 }  // namespace
 
+ClusterFamily::FlowInfo::~FlowInfo() = default;
+ClusterFamily::MigrationInfo::MigrationInfo(std::uint32_t flows_num, std::string ip, uint16_t port,
+                                            std::vector<ClusterConfig::SlotRange> slots,
+                                            Context::ErrHandler err_handler)
+    : host_ip(ip),
+      port(port),
+      slots(slots),
+      cntx(err_handler),
+      state(ClusterSlotMigration::State::C_CONNECTING),
+      flows(flows_num) {
+}
+
+void ClusterFamily::MigrationInfo::StartFlow(DbSlice* slice, uint32_t sync_id,
+                                             journal::Journal* journal, io::Sink* dest) {
+  state = ClusterSlotMigration::State::C_FULL_SYNC;
```

Comment:
The stat is for the entire migration right? and we set it here inside StartFlow which is executed on all connection, I think its a bit confusing.
Lets have a state per flow and if all flows and the global migration state will be the min state of all flows

Reply:
this was we also can remove streamer->IsStableSync() and having streamer updating the state of the flow once finish traversing

Reply:
we don't stop send journal data in streamer so we without this flag we don;t know that we in stable sync

Reply:
I can not remove streamer->IsStableSync() because I don't know when we stop sending existing data.